src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -86,25 +87,58 @@
       Logger.getLogger(DocumentTraverser.class.getName());
 
   private final ConfigOptions options;
-  private final int maxRecords;
+  private Checkpoint incrementalCheckpoint;
 
   public DocumentTraverser(ConfigOptions options) {
     this.options = options;
-    // Leave room for the continuation URLs.
-    this.maxRecords = options.getMaxFeedUrls() - 2;
+    Date yesterday = new Date(new Date().getTime() - TimeUnit.DAYS.toMillis(1));
+    this.incrementalCheckpoint =
+        new Checkpoint("incremental", yesterday , Id.ZERO_ID);
   }
 
   @Override
   public void getDocIds(Checkpoint checkpoint, DocIdPusher pusher)
       throws IOException, InterruptedException {
+    ArrayList<DocIdPusher.Record> records = new ArrayList<>();
+    // - 2 to leave room for the continuation URLs.
+    Checkpoint newCheckpoint =
+        getDocIds(checkpoint, false, records, options.getMaxFeedUrls() - 2);
+    // TODO(jlacey): if (records.size() == maxRecords)
+    if (newCheckpoint != checkpoint) {
+      records.add(
+          new DocIdPusher.Record.Builder(newDocId(newCheckpoint))
+          .setCrawlImmediately(true).build());
+    }
+    records.add(
+        new DocIdPusher.Record.Builder(newDocId(checkpoint))
+        .setDeleteFromIndex(true).build());
+    pusher.pushRecords(records);
+  }
+
+  @Override
+  public void getModifiedDocIds(DocIdPusher pusher)
+      throws IOException, InterruptedException {
+    ArrayList<DocIdPusher.Record> records = new ArrayList<>();
+    // Keep feeding full batches until we run out.
```

Comment:
We probably discussed this a long time ago, but why use internal batches here rather than continuation URLs? I came to think the use of continuation URLs was bordering on an affectation, but I'm not sure that having both mechanisms is an improvement. The continuations are likely more robust in probably unusual circumstances.